finality-aleph/src/party/backup.rs

```diff
@@ -118,27 +118,22 @@ pub fn rotate(
     Ok((backup_saver, backup_loader))
 }
 
-/// Removes the backup directory for a session.
+/// Removes the backup directory for all sessions except the current session.
 ///
 /// `backup_path` is the path to the backup directory (i.e. the argument to `--backup-saving-path`).
 /// If it is `None`, nothing is done.
 ///
-/// Any filesystem errors are logged and dropped.
+/// Any filesystem errors are returned.
 ///
-/// This should be done after the end of the session.
-pub fn remove(path: Option<PathBuf>, session_id: u32) {
-    let path = match path {
-        Some(path) => path.join(session_id.to_string()),
-        None => return,
-    };
-    match fs::remove_dir_all(path) {
-        Ok(()) => {
-            debug!(target: "aleph-party", "Removed backup for session {}", session_id);
-        }
-        Err(err) => {
-            if err.kind() != io::ErrorKind::NotFound {
-                warn!(target: "aleph-party", "Error cleaning up backup for session {}: {}", session_id, err);
+/// This should be done at the beginning of the new session.
+pub fn remove_old_backups(path: Option<PathBuf>, current_session: u32) -> io::Result<()> {
+    if let Some(path) = path {
+        for read_dir in fs::read_dir(path.clone())? {
+            let item = read_dir?;
```

Comment:
We shouldn't get errors there in practice, but theoretically `if let Ok(item) = read_dir) {}` is more correct there.

Reply:
Hm, I dunno, this logging an error on failure and not doing anything else seems good to me actually. Trying to power through with data deletion tends to be unnecessarily risky.

Reply:
I can agree, we can leave it as it is and not overcomplicate error handling when it doesn't give clear advantages

Reply:
ok, leaving as it is